dependency/vault_common.go

```diff
@@ -122,6 +121,16 @@ func leaseCheckWait(s *Secret) time.Duration {
 		}
 	}
 
+	// Handle if this is an AppRole secret_id with no lease
+	if _, ok := s.Data["secret_id"]; ok && s.LeaseID == "" {
+		if expInterface, ok := s.Data["secret_id_ttl"]; ok {
+			if ttlData, err := expInterface.(json.Number).Int64(); err == nil && ttlData > 0 {
+				base = int(ttlData) + 1
+				log.Printf("[DEBUG] Found secret_id and set lease duration to %d seconds", base)
```

Comment:
Slightly more descriptive message:

```suggestion
				log.Printf("[DEBUG] Found approle secret_id and non-zero secret_id_ttl, setting lease duration to %d seconds", base)
```